examples/vulkan/Vulkan.cpp

```diff
@@ -981,22 +981,20 @@ class VulkanExample
         attachmentDescriptions[1].initialLayout = VK_IMAGE_LAYOUT_UNDEFINED;
         attachmentDescriptions[1].finalLayout = VK_IMAGE_LAYOUT_DEPTH_STENCIL_ATTACHMENT_OPTIMAL;
 
-        VkAttachmentReference attachmentReferences[2];
+        VkAttachmentReference colorAttachmentReference = VkAttachmentReference();
+        colorAttachmentReference.attachment = 0;
+        colorAttachmentReference.layout = VK_IMAGE_LAYOUT_COLOR_ATTACHMENT_OPTIMAL;
 
-        attachmentReferences[0] = VkAttachmentReference();
-        attachmentReferences[0].attachment = 0;
-        attachmentReferences[0].layout = VK_IMAGE_LAYOUT_COLOR_ATTACHMENT_OPTIMAL;
-
-        attachmentReferences[1] = VkAttachmentReference();
-        attachmentReferences[1].attachment = 1;
-        attachmentReferences[1].layout = VK_IMAGE_LAYOUT_DEPTH_STENCIL_ATTACHMENT_OPTIMAL;
+        VkAttachmentReference depthStencilAttachmentReference = VkAttachmentReference();
```

Comment:
```suggestion
        VkAttachmentReference depthStencilAttachmentReference;
```

Reply:
This changes semantics, from zero-initialized to uninitialized.

Maybe 
```cpp
VkAttachmentReference depthStencilAttachmentReference{};
// or
VkAttachmentReference depthStencilAttachmentReference = {};
```
(the 2nd notation makes it more clear it's a C struct and not a constructor call)

Reply:
I think that this should be zero initialized in case some implementation have (or may have in the future) additional fields. I've changed both occurrences to `= {};`
Also rebased branch on top of master.